src/interpreter/Environment.ts

```diff
@@ -152,7 +156,11 @@ export class Environment {
         );
 
         if (source) {
-            return source.get(lowercaseName)!;
+            let variableToReturn = source.get(lowercaseName)!;
+            if (this.isMockedFunction(variableToReturn, lowercaseName)) {
```

Comment:
>

We could probably add an extra check to ensure we only mock names from the `module` and `global` scopes to prevent mocking of things in local scope!

```suggestion
            if (souce !== this.function && this.isMockedFunction(variableToReturn, lowercaseName)) {
```

Reply:
Can you provide an example brightscript snippet capturing the spirit of `prevent mocking of things in local scope!` ?

I am consistently seeing `this.function` as an empty map so the mock is always being returned.

Reply:
Sorry for the slow response — took the weekend off from side-projecting 😄 

It's mostly to prevent strange variable shadowing issues like:

```brightscript
sub main()
    print foo() ' => "Named Foo"

    foo = function()
        return "--inline foo--"
    end function

    print foo() ' => "--inline foo--"

    _brs_.mockFunction("foo", function()
        return "mOcKeD fOo"
    end function)

    print foo() ' => "mOcKeD fOo"
end sub

function foo()
    return "Named Foo"
end function
```

Unless I'm misunderstanding things!  That's absolutely possible 😅

Reply:
Is the snippet you posted the correct expected behavior? Just want to be super clear before I make any changes 👍

Reply:
Basically if I _DON'T_ implement the check against function scope the output is:
```
Named Foo
--inline foo--
mOcKeD fOo
```

But if I do implement the check against function scope I get:
```
Named Foo
--inline foo--
--inline foo--
```

Reply:
I suspect we do want to implement that check, and prevent mocking of a local variable `foo = function() …`.  If we find a use-case where we do want that, we can always remove the check.

But!  If you've got a strong use-case for allowing that I'm totally okay with that too.  We make the rules here 😜